updated nn packet-in protocol to request information from agent, impl…

antoninbas opened the pull request:
…emented method mac() for nn platform

antoninbas commented:
replaces #37 
- extends the msg protocol in a more general way
- more reliable
- re-use PTF code to read mac address

pavel-shirshov commented:
Antonin,

IMO It's better don't reuse ptf code inside of ptf_nn_agent. I prefer that ptf_nn_agent should be a portable script.We usually don't setup full ptf package on our switches to run ptf_nn_agent

pavel-shirshov commented:
Another thought:
You're extending the protocol "in a more general way", but restrict payload size by 32. Why 32? Not 64 or 16?
Why not use special message for any info request? There is enough integer number for any info request.

antoninbas commented:
@pavel-shirshov 
Thanks for your review.
I removed the `ptf.netutils` import from the agent.
As for your second comment, I am not sure I understand what you meant. The size of the request key was limited to 32 bytes (which I think was more than enough), the size of the reply was not limited. In any case, I replaced the string key by an integer. However, I will keep a single message type for requests, as it makes responding to unsupported requests easier. So the format for port info request messages is `<msg_type=MSG_TYPE_INFO_REQ(5)>|<port_number>|<request_info_type>`. For the mac address, `request_info_type` is `MSG_INFO_TYPE_HWADDR` which is `0`.

pavel-shirshov commented:
Hi Antonin,

Thank you for removing the dependency.
About my second point. I wasn't agree on the way you expanded the protocol. I thought it's overkill to introduce the get_info call with subcalls. But it doesn't matter honestly.

Thank you for the pull request.
